pkg/aws_config_client/completer.go

```diff
@@ -253,9 +297,18 @@ func (c *completer) Complete(out *ini.File) error {
 		return err
 	}
 
-	err = c.writeAWSProfiles(out, region, profiles)
+	newAWSProfiles, err := c.assembleAWSConfig(region, profiles)
 	if err != nil {
 		return err
 	}
-	return nil
+
+	mergedConfig, err := c.mergeConfigs(newAWSProfiles, base)
+	if err != nil {
+		return err
+	}
+	_, err = mergedConfig.WriteTo(fileDestination)
+	if err != nil {
+		return errors.Wrap(err, "Unable to merge old and new config files")
+	}
+	return errors.Wrapf(err, "Could not write new aws config.")
```

Comment:
```suggestion
	return errors.Wrapf(err, "Could not write new aws config.")
```
double error check there